Documentation/gettingstarted/cni-chaining-aws-cni.rst

```diff
@@ -6,69 +6,147 @@
-done.
+Ensure that the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_
+plugin is installed — which will already be the case if you have created an EKS
+cluster. Also, ensure the version of the plugin is up-to-date as per the above.
 
 .. include:: k8s-install-download-release.rst
 
-Deploy Cilium release via Helm:
+Deploy Cilium via Helm:
 
 .. parsed-literal::
 
    helm install cilium |CHART_RELEASE| \\
      --namespace kube-system \\
      --set cni.chainingMode=aws-cni \\
-     --set enableIPv4Masquerade=false \\
+     --set masquerade=false \\
      --set tunnel=disabled \\
-     --set nodeinit.enabled=true
+     --set nodeinit.enabled=true \\
+     --set endpointRoutes.enabled=true
 
-This will enable chaining with the aws-cni plugin. It will also disable
-tunneling. Tunneling is not required as ENI IP addresses can be directly routed
-in your VPC. You can also disable masquerading for the same reason.
+This will enable chaining with the AWS VPC CNI plugin. It will also disable
+tunneling, as it's not required since ENI IP addresses can be directly routed
+in the VPC. For the same reason, masquerading can be disabled as well.
 
 Restart existing pods
 =====================
 
-The new CNI chaining configuration will *not* apply to any pod that is already
-running in the cluster. Existing pods will be reachable and Cilium will
-load-balance to them but policy enforcement will not apply to them and
-load-balancing is not performed for traffic originating from existing pods.
-You must restart these pods in order to invoke the chaining configuration on
-them.
+The new CNI chaining configuration *will not* apply to any pod that is already
+running in the cluster. Existing pods will be reachable, and Cilium will
+load-balance *to* them, but not *from* them. Policy enforcement will also not
+be applied. For these reasons, you must restart these pods so that the chaining
+configuration can be applied to them.
+
+The following command can be used to check which pods need to be restarted:
 
-If you are unsure if a pod is managed by Cilium or not, run ``kubectl get cep``
-in the respective namespace and see if the pod is listed.
+.. code-block:: bash
+
+   for ns in $(kubectl get ns -o jsonpath='{.items[*].metadata.name}'); do
+        ceps=$(kubectl -n "${ns}" get cep \
+            -o jsonpath='{.items[*].metadata.name}')
+        pods=$(kubectl -n "${ns}" get pod \
+            -o custom-columns=NAME:.metadata.name,NETWORK:.spec.hostNetwork \
+            | grep -E '\s(<none>|false)' | awk '{print $1}' | tr '\n' ' ')
+        ncep=$(echo "${pods} ${ceps}" | tr ' ' '\n' | sort | uniq -u | paste -s -d ' ' -)
+        for pod in $(echo $ncep); do
+          echo "${ns}/${pod}";
+        done
+   done
 
 .. include:: k8s-install-validate.rst
 
+.. important::
+
+   As mentioned above, some advanced Cilium features may be limited when
+   chaining with other CNI plugins. Hence, some of the connectivity tests
+   above may fail.
```

Comment:
In this case, I think it should be better to give the user the specific `cilium connectivity test` command [with exceptions for tests that are supposed to fail](https://github.com/cilium/cilium-cli/blob/master/internal/cli/cmd/connectivity.go#L79) with CNI chaining on AWS. For example, to skip `pod-to-local-nodeport` test:

```
cilium connectivity test --test '!pod-to-local-nodeport'
```

Reply:
Oh cool — I'm not sure if we know exactly what tests may fail, but I'll try to check.

Reply:
No tests seem to fail as of `1.10.0-rc1` and CLI `0.6`. I have removed the note, I guess we can always add it back if we get failing tests in the future.